plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt

```diff
@@ -56,55 +67,62 @@ class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
             object : RefreshQChatPanelButtonPressedListener {
                 override fun onRefresh() {
                     runInEdt {
-                        contentManager.removeAllContents(true)
-                        prepareChatContent(project, contentManager)
+                        prepareChatContent(project)
                     }
                 }
             }
         )
 
+        project.messageBus.connect().subscribe(
+            ShowCriticalNotificationBannerListener.TOPIC,
+            object : ShowCriticalNotificationBannerListener {
+                override fun onReceiveEmergencyNotification(title: String, message: String, actions: List<NotificationFollowupActions>?) {
+                    notificationPanel.updateNotificationPanel(title, message, actions)
+                }
+            }
+        )
```

Comment:
message bus is overkill, just have the notification service expose a listener pattern